src/main/groovy/com/blackducksoftware/integration/hub/detect/DetectProjectManager.groovy

```diff
@@ -193,27 +197,39 @@ class DetectProjectManager implements SummaryResultReporter {
         return detectSummaryResults;
     }
 
-    private String createBdioFilename(BomToolType bomToolType, String finalSourcePathPiece, String projectName, String projectVersionName) {
-        def names = [finalSourcePathPiece, projectName, projectVersionName]
-        names.sort { -it.size() }
-        String filename = generateFilename(bomToolType, finalSourcePathPiece, projectName, projectVersionName)
-        for (int i = 0; (filename.length() >= 255) && (i < 3); i++) {
-            names[i] = DigestUtils.sha1Hex(names[i])
-            if (names[i].length() > 15) {
-                names[i] = names[i].substring(0, 15)
-            }
+    private String generateShortenedFilename(BomToolType bomToolType, String finalSourcePathPiece, ExternalId externalId) {
+
+        List<String> filenamePieces = new ArrayList<>();
+        filenamePieces.addAll(externalId.getExternalIdPieces());
+        filenamePieces.add(finalSourcePathPiece);
+        String filename = generateFilename(bomToolType, filenamePieces)
 
-            filename = generateFilename(bomToolType, names[0], names[1], names[2])
+        if (filename.length() >= 255){
```

Comment:
Nice - sorting this direction makes it easier to have an arbitrary number of pieces. Solid.